--- gwibber/microblog/dispatcher.py_orig	2011-04-06 17:00:31.314746403 +0200
+++ gwibber/microblog/dispatcher.py	2011-04-06 17:02:14.058437073 +0200
@@ -761,12 +761,14 @@
 
     self.has_nm = None
 
-    try:
-      self.nm = self.sysbus.get_object(NM_DBUS_SERVICE, NM_DBUS_OBJECT_PATH)
-      self.nm.connect_to_signal("StateChanged", self.on_connection_changed)
-      self.has_nm = True
-    except:
-      pass
+    #disable NetworkManager until I know how the new 
+    # NM handles states
+    #try:
+    #  self.nm = self.sysbus.get_object(NM_DBUS_SERVICE, NM_DBUS_OBJECT_PATH)
+    #  self.nm.connect_to_signal("StateChanged", self.on_connection_changed)
+    #  self.has_nm = True
+    #except:
+    #  pass
 
   def on_connection_changed(self, state):
     log.logger.debug("Network state changed, new state is %d", state)
