--- configure.ac.old	2010-02-26 18:31:46.920805847 +0100
+++ configure.ac	2010-02-26 18:32:22.355223285 +0100
@@ -40,7 +40,7 @@
 GEOCLUE_REQUIRED=0.11
 WEBKIT_REQUIRED=1.1.7
 KEYRING_REQUIRED=2.22
-NETWORK_MANAGER_REQUIRED=0.7.0
+NETWORK_MANAGER_REQUIRED=0.8
 
 # Use --enable-maintainer-mode to disabled deprecated symbols
 GNOME_MAINTAINER_MODE_DEFINES
@@ -184,7 +184,7 @@
 
    PKG_CHECK_MODULES(NETWORK_MANAGER,
    [
-      libnm_glib >= $NETWORK_MANAGER_REQUIRED
+      libnm-glib >= $NETWORK_MANAGER_REQUIRED
    ], have_nm="yes", have_nm="no")
 
    if test "x$have_nm" = "xyes"; then
