Index: libswscale/swscale.c
===================================================================
--- libswscale/swscale.c	(revision 32040)
+++ libswscale/swscale.c	(working copy)
@@ -1261,16 +1261,25 @@
 
 #if ARCH_X86
     // ordered per speed fastest first
+#if COMPILE_TEMPLATE_MMX2
     if (flags & SWS_CPU_CAPS_MMX2) {
         sws_init_swScale_MMX2(c);
         return swScale_MMX2;
-    } else if (flags & SWS_CPU_CAPS_3DNOW) {
+    } else
+#endif
+#if COMPILE_TEMPLATE_AMD3DNOW
+    if (flags & SWS_CPU_CAPS_3DNOW) {
         sws_init_swScale_3DNow(c);
         return swScale_3DNow;
-    } else if (flags & SWS_CPU_CAPS_MMX) {
+    } else
+#endif
+#if COMPILE_TEMPLATE_MMX
+    if (flags & SWS_CPU_CAPS_MMX) {
         sws_init_swScale_MMX(c);
         return swScale_MMX;
-    } else {
+    } else
+#endif
+    {
         sws_init_swScale_C(c);
         return swScale_C;
     }
