--- suds/bindings/__init__.py.old	2009-06-08 21:31:55.000000000 +0200
+++ suds/bindings/__init__.py	2009-06-08 21:37:41.000000000 +0200
@@ -25,11 +25,14 @@
     @ivar lang: The string language when set (may be None).
     @type lang: str
     """
+    __slots__ = ('lang',)
+
     @classmethod
     def string(cls, s, lang=None):
-        xls = xlstr(s)
-        xls.lang = lang
-        return xls
-    def __init__(self, s):
-        unicode.__init__(self, s)
-        self.lang = None
\ No newline at end of file
+	    return xlstr(s, lang=lang)
+
+    def __new__(cls, *args, **kwargs):
+	    lang = kwargs.pop('lang', None)
+	    res = super(xlstr, cls).__new__(cls, *args, **kwargs)
+	    res.lang = lang
+	    return res
